packages/flame/lib/src/gestures/events.dart

```diff
@@ -0,0 +1,181 @@
+
+  late final position = EventPosition(
+    _game,
+    _position,
+    _globalPosition,
+  );
+
+  PositionInfo(
+    this._game,
+    this._position,
+    this._globalPosition,
+    T raw,
+  ) : super(raw);
+}
+
+class TapDownInfo extends PositionInfo<TapDownDetails> {
+  TapDownInfo.fromDetails(
+    Game game,
+    TapDownDetails raw,
+  ) : super(game, raw.localPosition, raw.globalPosition, raw);
+}
+
+class TapUpInfo extends PositionInfo<TapUpDetails> {
+  TapUpInfo.fromDetails(
+    Game game,
+    TapUpDetails raw,
+  ) : super(game, raw.localPosition, raw.globalPosition, raw);
+}
+
+class LongPressStartInfo extends PositionInfo<LongPressStartDetails> {
+  LongPressStartInfo.fromDetails(
+    Game game,
+    LongPressStartDetails raw,
+  ) : super(game, raw.localPosition, raw.globalPosition, raw);
+}
+
+class LongPressEndInfo extends PositionInfo<LongPressEndDetails> {
+  late final Vector2 velocity = _game.scaleOffset(raw.velocity.pixelsPerSecond);
+
+  LongPressEndInfo.fromDetails(
+    Game game,
+    LongPressEndDetails raw,
+  ) : super(game, raw.localPosition, raw.globalPosition, raw);
+}
+
+class LongPressMoveUpdateInfo extends PositionInfo<LongPressMoveUpdateDetails> {
+  LongPressMoveUpdateInfo.fromDetails(
+    Game game,
+    LongPressMoveUpdateDetails raw,
+  ) : super(game, raw.localPosition, raw.globalPosition, raw);
+}
+
+class ForcePressInfo extends PositionInfo<ForcePressDetails> {
+  late final double pressure = raw.pressure;
+
+  ForcePressInfo.fromDetails(
+    Game game,
+    ForcePressDetails raw,
+  ) : super(game, raw.localPosition, raw.globalPosition, raw);
+}
+
+class PointerScrollInfo extends PositionInfo<PointerScrollEvent> {
+  late final Vector2 scrollDelta = _game.scaleOffset(raw.scrollDelta);
+
+  PointerScrollInfo.fromDetails(
+    Game game,
+    PointerScrollEvent raw,
+  ) : super(game, raw.localPosition, null, raw);
+}
+
+class PointerHoverInfo extends PositionInfo<PointerHoverEvent> {
+  PointerHoverInfo.fromDetails(
+    Game game,
+    PointerHoverEvent raw,
+  ) : super(game, raw.localPosition, null, raw);
+}
+
+class DragDownInfo extends PositionInfo<DragDownDetails> {
+  DragDownInfo.fromDetails(
+    Game game,
+    DragDownDetails raw,
+  ) : super(game, raw.localPosition, raw.globalPosition, raw);
+}
+
+class DragStartInfo extends PositionInfo<DragStartDetails> {
+  DragStartInfo.fromDetails(
+    Game game,
+    DragStartDetails raw,
+  ) : super(game, raw.localPosition, raw.globalPosition, raw);
+}
+
+class DragUpdateInfo extends PositionInfo<DragUpdateDetails> {
+  late final Vector2 delta = _game.scaleOffset(raw.delta);
+
+  DragUpdateInfo.fromDetails(
+    Game game,
+    DragUpdateDetails raw,
+  ) : super(game, raw.localPosition, raw.globalPosition, raw);
+}
+
+class DragEndInfo extends BaseInfo<DragEndDetails> {
+  final Game _game;
+  late final Vector2 velocity = _game.scaleOffset(raw.velocity.pixelsPerSecond);
+  double? get primaryVelocity => raw.primaryVelocity;
```

Comment:
this should be an EventPosition right?

Reply:
similarly below

Reply:
DragEndDetails don't have coordiantes.

Reply:
it has the velocity, instead of having velocityLocal, velocityGlobal, velocityGame, it should use the wrapper class